sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/ApplicationServiceImpl.java

```diff
@@ -34,58 +37,74 @@
 
   @Override
   public Application create(String realm, Application application) {
-    sugoiEventPublisher.publishCustomEvent(
-        realm,
-        null,
-        SugoiEventTypeEnum.CREATE_APPLICATION,
-        Map.ofEntries(Map.entry(EventKeysConfig.APPLICATION, application)));
-    return storeProvider.getWriterStore(realm).createApplication(application);
+    if (!findById(realm, application.getName()).isPresent()) {
+      String appName = storeProvider.getWriterStore(realm).createApplication(application).getName();
+      sugoiEventPublisher.publishCustomEvent(
+          realm,
+          null,
+          SugoiEventTypeEnum.CREATE_APPLICATION,
+          Map.ofEntries(Map.entry(EventKeysConfig.APPLICATION, application)));
+      return findById(realm, appName)
```

Comment:
I find it weird to throw an ApplicationNotFoundException here. If the app does not exist, this means we could not create it, it should have thrown another exception I think.